packages/fxa-payments-server/src/components/CouponForm/index.test.tsx

```diff
@@ -28,6 +50,38 @@ describe('CouponForm', () => {
   });
 
   it('shows an error message when invalid coupon code is used', async () => {
+    const mockSetCoupon = jest.fn();
+    const subject = () => {
+      return render(
+        <CouponForm
+          planId={SELECTED_PLAN.plan_id}
+          coupon={undefined}
+          setCoupon={mockSetCoupon}
+        />
+      );
+    };
+
+    const { queryByTestId, getByTestId } = subject();
+    fireEvent.change(getByTestId('coupon-input'), { target: { value: 'a' } });
+    fireEvent.click(getByTestId('coupon-button'));
+
+    await waitForExpect(() => {
+      expect(queryByTestId('coupon-error')).toBeInTheDocument();
+      expect(mockSetCoupon).toBeCalledWith(undefined);
+    });
+
+    const couponError = getByTestId('coupon-error');
+
+    expect(couponError).toHaveTextContent(
+      'The code you entered is invalid or expired.'
+    );
+  });
```

Comment:
Thanks for adding this test.  Doesn't seem to be related to zero charge invoices, but 👍 anyway.

Reply:
It's coming up in my next PR, so I thought might as well add it in here as well. :)